app/src/main/java/org/fossasia/pslab/communication/ScienceLab.java

```diff
@@ -872,11 +872,15 @@ public void captureHighResolutionTraces(String channel, int samples, double time
     }
 
     public HashMap<String, double[]> fetchTrace(int channelNumber) {
-        this.fetchChannel(channelNumber);
-        HashMap<String, double[]> retData = new HashMap<>();
-        retData.put("x", this.aChannels.get(channelNumber - 1).getXAxis());
-        retData.put("y", this.aChannels.get(channelNumber - 1).getYAxis());
-        return retData;
+        try {
+            this.fetchChannel(channelNumber);
+            HashMap<String, double[]> retData = new HashMap<>();
+            retData.put("x", this.aChannels.get(channelNumber - 1).getXAxis());
+            retData.put("y", this.aChannels.get(channelNumber - 1).getYAxis());
+            return retData;
+        } catch (IndexOutOfBoundsException e) {
```

Comment:
Hi @viveksb007 ! 
`this.aChannels` is empty when the device is not connected and throws the `IndexOutOfBoundException` when it tries to get values from it. Since there are three implementations calling functions to fetch values from `aChannels`, I caught the exception at this point and returned a null object so that in the background task it can be identified as an error and  proceed no further. Is that a good approach?